tests/core/asyncify_advise.c

```diff
@@ -0,0 +1,29 @@
+#include <emscripten.h>
+
+int test = 0;
+
+EM_JS(void, async_func, (), {
+  Asyncify.handleSleep(wakeUp => {
+    setTimeout(wakeUp, 10);
+  });
+});
+
+void i() { async_func(); test++; }
+
+void h() { test++; }
+
+void g() { h(); i(); test++; }
+
+void f() { test++; }
+
+void e() { async_func(); test++; }
+
+void d() { test++; }
+
+void c() { d(); e(); test++; }
+
+void b() { test++; }
+
+void a() { b(); c(); test++; }
```

Comment:
Yeah, it is a little tricky to avoid the LLVM optimizer inlining such little functions and breaking the test. That is a real problem with this feature though, as the call stacks do change with optimizations. So I actually wonder if it's enough to run this test in `-O0` (and move it to `tests/test_other.py` since it would no longer need the special logic in `core` for running in all modes).

That is, it may be best in general for people to use `ASYNCIFY_ADVISE` with `-O0`, which will give the full call stack (which may not all be needed if optimizations remove some).

It that makes sense, it may also be good to mention that in the docs for the option?